lib/rules/arrow-linebreak.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @fileoverview enforce the location of arrow function bodies
+ * @author Sharmila Jesupaul
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+module.exports = {
+    meta: {
+        docs: {
+            description: "enforce the location of arrow function bodies",
+            category: "Stylistic Issues",
+            recommended: false
+        },
+        fixable: "whitespace",
+        schema: [
+            {
+                enum: ["beside", "below"]
+            }
+        ]
+    },
+
+    create(context) {
+        const sourceCode = context.getSourceCode();
+
+        //----------------------------------------------------------------------
+        // Helpers
+        //----------------------------------------------------------------------
+        /**
+         * Gets the applicable preference for a particular keyword
+         * @returns {string} The applicable option for the keyword, e.g. 'beside'
+         */
+        function getOption() {
+            return context.options[0] || "beside";
+        }
+
+        /**
+         * Validates the location of an arrow function body
+         * @param {ASTNode} node The arrow function body
+         * @param {string} keywordName The applicable keyword name for the arrow function body
+         * @returns {void}
+         */
+        function validateStatement(node) {
+            const option = getOption();
+            const bodyNode = node.body;
+
+            let tokenBefore = sourceCode.getTokenBefore(bodyNode);
+
+            const hasParens = tokenBefore.value === "(";
+            const parensWithLinebreak = hasParens && tokenBefore.loc.end.line < bodyNode.loc.start.line;
+
+            if (node.type === "BlockStatement" || parensWithLinebreak) {
+                return;
+            }
+
+            let fixerTarget = bodyNode;
+
+            if (hasParens) {
+                fixerTarget = tokenBefore;
+                tokenBefore = sourceCode.getTokenBefore(fixerTarget);
```

Comment:
I think this won't be correct if the arrow function body is surrounded by multiple sets of parentheses:

```js
var x = foo =>
    ((((bar))));
```

In that case, `tokenBefore` will refer to the wrong thing because it will still be an opening paren.

Instead, one option would be to use some thing like

```js
sourceCode.getTokenBefore(node.body, token => token.value !== "(")
```

...which will get the first token before the body that isn't an opening paren.

Reply:
Good call, I completely missed this case entirely 😔 Will make an update soon.